TestGocrossWrapper fails when run inside a `git worktree`

### What is the issue?

Until Go 1.27 is released with the fix for golang/go#58218, [TestGocrossWrapper](https://github.com/tailscale/tailscale/blob/211ef67222075eb72f3c560d83a693d122b49843/tool/gocross/gocross_wrapper_test.go#L15-L27) fails when run inside a [git linked worktree](https://man7.org/linux/man-pages/man1/git-worktree.1.html). 

### Steps to reproduce

``` console
sfllaw@hostname:~/src/tailscale$ git worktree add --checkout ../tailscale-worktree
Preparing worktree (new branch 'tailscale-worktree')
HEAD is now at 211ef6722 tailcfg,ipn/ipnlocal: regulate netmap caching via a node attribute (#19117)
sfllaw@hostname:~/src/tailscale$ cd ../tailscale-worktree
sfllaw@hostname:~/src/tailscale-foo$ ./tool/go test ./tool/gocross
--- FAIL: TestGocrossWrapper (0.46s)
    gocross_wrapper_test.go:24: expected to find 'gocross_ok=1'; got output:
        + [[ linux-gnu == \c\y\g\w\i\n ]]
        + [[ linux-gnu == \m\s\y\s ]]
        + [[ '' == \1 ]]
        + go_toolchain_rev_file=go.toolchain.rev
        + repo_root=./../..
        + cd ./../..
        + toolchain=
        + read -r REV
        + case "$REV" in
        + toolchain=/home/sfllaw/.cache/tsgo/179b46cade24e44f53b53a3cbcc7b7eb78469c31
        + [[ ! -f /home/sfllaw/.cache/tsgo/179b46cade24e44f53b53a3cbcc7b7eb78469c31.extracted ]]
        + '[' 1 '!=' 1 ']'
        + [[ -d /home/sfllaw/.cache/tsgo/179b46cade24e44f53b53a3cbcc7b7eb78469c31 ]]
        ++ grep -E '^go ' go.mod
        ++ cut -f2 -d.
        + want_go_minor=26
        + have_go_minor=
        + [[ -f /home/sfllaw/.cache/tsgo/179b46cade24e44f53b53a3cbcc7b7eb78469c31/VERSION ]]
        ++ head -1 /home/sfllaw/.cache/tsgo/179b46cade24e44f53b53a3cbcc7b7eb78469c31/VERSION
        ++ cut -f2 -d.
        + have_go_minor=26
        + have_go_minor=26
        + [[ -z 26 ]]
        + [[ 26 -lt 26 ]]
        + gocross_path=./gocross
        + gocross_ok=0
        ++ git rev-parse HEAD
        + wantver=211ef67222075eb72f3c560d83a693d122b49843
        + [[ -x ./gocross ]]
        ++ ./gocross gocross-version
        ++ echo ''
        + gotver=
        + [[ '' == \2\1\1\e\f\6\7\2\2\2\0\7\5\e\b\7\2\f\3\c\5\6\0\d\8\3\a\6\9\3\d\1\2\2\b\4\9\8\4\3 ]]
        + [[ 0 == \0 ]]
        + unset GOOS
        + unset GOARCH
        + unset GO111MODULE
        + unset GOROOT
        + export CGO_ENABLED=0
        + CGO_ENABLED=0
        + /home/sfllaw/.cache/tsgo/179b46cade24e44f53b53a3cbcc7b7eb78469c31/bin/go build -o ./gocross -ldflags '-X tailscale.com/version.gitCommitStamp=211ef67222075eb72f3c560d83a693d122b49843' tailscale.com/tool/gocross
        + repo_root=./../..
        + unset GOROOT
        + '[' 1 '!=' 1 ']'
        + exec ./../../gocross version
        go version go1.26.1 linux/amd64
FAIL
FAIL	tailscale.com/tool/gocross	0.533s
FAIL
```

--- a/tool/gocross/gocross_wrapper_test.go
+++ b/tool/gocross/gocross_wrapper_test.go
@@ -6,13 +6,26 @@
 package main
 
 import (
+	"bytes"
+	"go/version"
 	"os"
 	"os/exec"
+	"runtime"
 	"strings"
 	"testing"
+
+	"tailscale.com/util/must"
 )
 
 func TestGocrossWrapper(t *testing.T) {
+	if version.Compare(runtime.Version(), "go1.27") < 0 {
+		gitDir := must.Get(exec.Command("git", "rev-parse", "--git-dir").Output())
+		gitCommonDir := must.Get(exec.Command("git", "rev-parse", "--git-common-dir").Output())
+		if !bytes.Equal(gitDir, gitCommonDir) {
+			t.Skip("skipping within git worktree, see https://go.dev/issue/58218")
+		}
+	}
+
 	for i := range 2 { // once to build gocross; second to test it's cached
 		cmd := exec.Command("./gocross-wrapper.sh", "version")
 		cmd.Env = append(os.Environ(), "CI=true", "NOBASHDEBUG=false", "TS_USE_GOCROSS=1") // for "set -x" verbosity

--- a/tool/gocross/gocross_wrapper_windows_test.go
+++ b/tool/gocross/gocross_wrapper_windows_test.go
@@ -4,13 +4,26 @@
 package main
 
 import (
+	"bytes"
+	"go/version"
 	"os"
 	"os/exec"
+	"runtime"
 	"strings"
 	"testing"
+
+	"tailscale.com/util/must"
 )
 
 func TestGocrossWrapper(t *testing.T) {
+	if version.Compare(runtime.Version(), "go1.27") < 0 {
+		gitDir := must.Get(exec.Command("git", "rev-parse", "--git-dir").Output())
+		gitCommonDir := must.Get(exec.Command("git", "rev-parse", "--git-common-dir").Output())
+		if !bytes.Equal(gitDir, gitCommonDir) {
+			t.Skip("skipping within git worktree, see https://go.dev/issue/58218")
+		}
+	}
+
 	for i := range 2 { // once to build gocross; second to test it's cached
 		cmd := exec.Command("pwsh", "-NoProfile", "-ExecutionPolicy", "Bypass", ".\\gocross-wrapper.ps1", "version")
 		cmd.Env = append(os.Environ(), "CI=true", "NOPWSHDEBUG=false", "TS_USE_GOCROSS=1") // for Set-PSDebug verbosity

